pkg/helpers/newapp/app/dockerimagelookup.go

```diff
@@ -331,6 +332,18 @@ func (s ImageImportSearcher) Search(precise bool, terms ...string) (ComponentMat
 		}
 
 		dockerImage, ok := image.Image.DockerImageMetadata.Object.(*dockerv10.DockerImage)
+
+		// When a ManifestListed Image is used the nested DockerConfig objects in a DockerImageManifest are not propagated
+		// The code marshalls the first entry's config, and assigns it to the top-level DockerImage
+		if len(image.Image.DockerImageManifests) != 0 && dockerImage.Config == nil {
+			conf := image.Manifests[0].DockerImageConfig
+			dc := &docker10.DockerImage{}
+			if err := json.Unmarshal([]byte(conf), &dc); err != nil {
+				// Errors do not cause issues in unmarshalling, and it is intentional, and logged out.
+				klog.V(4).Infof("unmarshalling failed for manifest listed image's dockerConfig: %#v %#v", conf, err)
+			}
+			dockerImage.Config = dc.Config
+		}
```

Comment:
I think this block should go after the `if !ok` check that's currently after it - in the `ok == false` case I think `dockerImage.Config` will be dereferencing a `nil` pointer.  Otherwise LGTM.

Reply:
oh good catch - I will take care of it.

Reply:
This looks like it's still going to be accessing fields of `dockerImage` before testing that `ok` is true.

Reply:
Hi Nalin, Hopefully I addressed your concerns above. thanks, Paul

Reply:
Yes, that should prevent attempts to access fields in a `nil` `dockerImage` value when `ok` is false.

Reply:
Thanks for the diligent review